[Magic Firewall] Added best practice content (#3323)
cloudflare/cloudflare-docs: products/magic-firewall/src/content/best-practices/extended-ruleset.md

---
title: Extended ruleset

> pcx-content-type: tutorial

Reviewer: Think this falls more under a `configuration` guide personally.

Author: Agreed

order: 2
---

# Extended suggested ruleset

For the best protection, replicate your current perimeter security ingress policy within Magic Firewall. If you are unable to export your current perimeter firewall rules, contact your Implementation Manager for help translating the rules into Magic Firewall rules.

If you are unable to export your current perimeter firewall rules, consider identifying categories of systems or user-groups that reside on your MT prefixes, for example:
- [Endpoints (user devices)](#endpoints-user-devices)

> - [Internal routers](#internal-routerfirewall-ips)

Reviewer: ```suggestion
- [Internal routers](#internal-routerfirewall-ip-addresses)
```

- [Web servers](#web-servers)
- [Non-web servers](#non-web-servers)

For each item above, consider the requirements in terms of their permitted Internet access. For example, permit what is required for legitimate traffic and block the rest.

## Create lists for using Magic Firewall rules

For more information on lists, refer to [Use rule lists](/how-to/use-rules-list).

You can also create a list from the dashboard from **Configurations** > **Lists** on your **Account Home**.

## Endpoints (User devices)

Endpoint devices do not operate as servers, which means:

> - They receive traffic from standard common ports, for example 80,443, towards their ephemeral ports, above 32768 in modern operating systems (1025 in older XP).

Reviewer: ```suggestion
- They receive traffic from standard common ports — for example 80 or 443 — towards their ephemeral ports, above 32768 in modern operating systems (1025 in older XP).
```

Reviewer: Figured having dashes helps break up the comma logjam a bit.

- Connections flow outwards, not inwards, and therefore do not receive TCP SYN or ACK packets.
- They typically only need client TCP and UDP, with no requirement for ingress ICMP.

For example, you can create a list for the combination of generic client TCP and client UDP that allows external pings or traceroutes and a catchall rule for all other protocols and traffic.

Create a list named **Endpoints** and specify the list of endpoints or user IP addresses to reference within the rules.

<Aside type="note">

Rule 10 in the example ruleset below is acting as a catch-all to block all traffic not permitted in rules 1-3 towards your list of Endpoint IP addresses. If you want to permit other traffic to these destination IP addresses, the new rule must be added before rule 10.

</Aside>

### Suggested rules

**Rule ID**: 1 <br/> 

> **Description**: Endpoints (clients) will receive traffic destined for ephemeral ports. Blocks inbound SYN-only traffic. (i.e. SYN-ACKs are permitted) <br/> 

Reviewer: ```suggestion
**Description**: Endpoints (clients) will receive traffic destined for ephemeral ports. Blocks inbound SYN-only traffic. (meaning SYN-ACKs are permitted) <br/> 
```

> **Match**: `ip.proto eq "tcp" and ip.dst in $endpoints and tcp.dstport in {32768..60999} and not (tcp.flags.syn and not tcp.flags.ack` <br/>

Reviewer: Are you missing a parentheses somewhere in this rule? There's one around `(tcp.flags.syn` but no closing one?

Author: Yes! Good eye, thank you for catching that.

**Action**: Allow <br/>

**Rule ID**: 2 <br/> 
**Description**: Endpoints (clients) will receive traffic destined for ephemeral ports <br/> 
**Match**: `ip.proto eq "udp" and ip.dst in $endpoints and udp.dstport in {32768..60999}` <br/>
**Action**: Allow <br/>

**Rule ID**: 3 <br/> 
**Description**: Permits ICMP traffic to destination IP addresses in `$endpoints` list with ICMP Types:
- Type 0 = Echo Reply <br/>
- Type 3 = Destination Unreachable <br/>
- Type 11 = Time Exceeded <br/>

**Match**: `ip.proto eq "icmp" and ip.dst in $endpoints and (icmp.type eq 0 or icmp.type eq 3 or icmp.type eq 11)` <br/>
**Action**: Allow <br/>

**Rule ID**: 10 <br/> 
**Description**: Otherwise deny all traffic to IP’s in `$endpoints` list <br/> 
**Match**: `ip.dst in $endpoints` <br/>
**Action**: Block <br/>

## Internal router/Firewall IP addresses

Follow the best practices for internal routers or firewall interface IP addresses on your MT prefixes below.

> 1. Create a list, **Internal routers** for example, list your IP addresses.

Reviewer: ```suggestion
1. Create [a list](https://developers.cloudflare.com/firewall/cf-dashboard/rules-lists), **Internal routers** for example, list your IP addresses.
```

1. Block ICMP if it is not needed.
1. Permit GRE/ESP as needed if the devices have GRE/IPSEC tunnels via the Internet.

### Suggested rules

**Rule ID**: 1 <br/> 

> **Description**: Permit limited ICMP traffic inbound: examples here are 

Reviewer: Confused on the formatting related to `examples here are`, is this phrase needed?

Author: Fixed

- Type 0 - Echo Reply
- Type 3 - Destination Unreachable
- Type 8 - Echo 

> - Type 11 = Time Exceeded <br/>

Reviewer: ```suggestion
- Type 11 - Time Exceeded <br/>
```

**Match**: `ip.proto eq "icmp" and ip.dst in $internal_routers and ( (icmp.type eq 0 or icmp.type eq 3) or (icmp.type eq 11) or (icmp.type eq 8) )` <br/>
**Action**: Allow<br/>

**Rule ID**: 2 <br/> 
**Description**: Block all other traffic destined to these IP addresses <br/> 
**Match**: `ip.dst in $internal_routers` <br/>
**Action**: Block <br/>

## Web Servers

Web servers require careful consideration of necessary traffic flows. Traffic for the **web server** functionality is required in addition to traffic flows where the web server is acting as a client. 

Where possible, permit the required destination IP addresses and ports for web servers and block everything else. Additional services, for example NTP/DNS, may be required along with the ports for the web traffic. 

> The following is an example of suggested rules, but you should only make changes based on your specific requirements. For example, if you are not proxied by Cloudflare Layer 7 protection, and you expect traffic sourced from the web towards your web servers:

Reviewer: ```suggestion
The following is an example of suggested rules, but you should only make changes based on your specific requirements. For example, if you are not proxied by Cloudflare Layer 7 protection and you expect traffic sourced from the web towards your web servers:
```

> 1. Create a list, **web servers** for example, to list IP addresses for your web servers.

Reviewer: ```suggestion
1. Create [a list](https://developers.cloudflare.com/firewall/cf-dashboard/rules-lists), **web servers** for example, to list IP addresses for your web servers.
```

1. Permit traffic for the web server traffic inbound from the Internet.
1. Permit traffic for the infrastructure or client traffic flows from the Internet, for example DNS and NTP.
1. Block all other traffic destined for the web server IP addresses.

### Suggested rules

**Rule ID**: 1 <br/> 
**Description**: Allows inbound HTTP/S traffic from the internet with SYN-only or ACK-only flag (not SYNACKs) <br/> 
**Match**: `ip.proto eq "tcp" and tcp.srcport in {32768..60999} and ip.dst in $web_servers and tcp.dstport in {80, 443} and not (tcp.flags.syn and tcp.flags.ack)` <br/>
**Action**: Allow <br/>

**Rule ID**: 2 <br/> 
**Description**: Allows UDP replies for DNS and NTP to web servers <br/> 
**Match**: `ip.dst in $web_servers and ip.proto eq “udp” and udp.srcport in {53 123} and udp.dstport in {1024..65535}` <br/>
**Action**: Allow if necessary but Disable if under attack <br/>

**Rule ID**: 3 <br/> 
**Description**: Catch-all to block all other traffic destined for web server IP addresses <br/> 
**Match**: `ip.dst in $web_servers` <br/>
**Action**: Block <br/>

> Alternatively, if you have Cloudflare Layer 7 protection, the Cloudflare Public IP addresses can be permitted as the source IP addresses to the destination IP addresses for the HTTP/HTTPS inbound traffic. This recommendation effectively replaces Rule 1 in the example above.

Reviewer: I would just move this section down into the next header.

Additionally, I would recommend you just tell them to create an IP list with Cloudflare's IP addresses and then reference that fake rule instead of listing out all our public IP addresses. Thinking that might be better b/c a) it's the recommended solution and b) our IP addresses might change in the future.

Author: I'm going to make note of this and address it in a separate PR so I can get formatting correct.
